pkg/api/v1alpha1/fluxconfig.go

```diff
@@ -71,10 +77,26 @@ func validateGitProviderConfig(config GitProviderConfig) error {
 	if len(config.RepositoryUrl) <= 0 {
 		return errors.New("'repositoryUrl' is not set or empty in gitProviderConfig; repositoryUrl is a required field")
 	}
+
 	err := validateRepositoryUrl(config.RepositoryUrl)
+	if err != nil {
+		return err
+	}
+
+	err = validateAuthentication()
 	return err
 }
 
+func validateAuthentication() error {
+	password, _ := git.GetTokenFromEnv(EksaGitPasswordTokenEnv)
+	privateKeyFilePath, _ := git.GetTokenFromEnv(EksaGitPrivateKeyTokenEnv)
```

Comment:
Would it possible to read this before hand and inject it as the rest of the dependencies in the constructor?

Reply:
@mitalipaygude what do you think about this?